### DISCUSSION TITLE
In branch deployment, how to materialize only assets whose definitions have changed

### DISCUSSION THREAD

**Discussion by littleK0i** [dagster-io/dagster#18758] [Q&A] [answered]
Let's say I have a git repo defining 100+ assets and a lot of dependencies between them. I work on ticket to change definition of 5 assets. Everything else remains the same.

When I commit changes, push git branch for testing and create a new branch deployment, is there any way to run only jobs for changed assets and upstream assets?

Basically, instead of running full ETL test, which may take a while and spend a lot of credits, I would like to test only assets which were actually affected by my change. It should help to save a lot of time & money, especially for very large repositories.

I had this feature in my own ETL tool, and now I am looking for a good way to do the same thing in Dagster. Ideally, without patching dagster core.

Thank you!

**Comment by sryza** [comment, marked as answer]
@littleK0i there currently isn't a straightforward way of doing this, although it's functionality we're interested in adding in the future and have been discussing the best way to support.  Currently, you can attach code versions to your assets and the Dagster UI can tell you what assets have code versions that are different than the code versions used for their latest materialization.  However, information about their latest materialization in the production deployment isn't yet available inside branch deployments, so you would need to find some other way of accessing it.
